modules/telestion-api/src/main/java/org/telestion/api/message/JsonMessage.java

```diff
@@ -117,4 +99,28 @@ default String className() {
 	default JsonObject json() {
 		return JsonObject.mapFrom(this);
 	}
+
+	/**
+	 * Reflection part of {@link #on(Class, Message, Handler)} and {@link #on(Class, Object, Handler)}.
+	 *
+	 * @param clazz			Class to cast to
+	 * @param jsonObject	Object to be casted
+	 * @param handler		Handler for the message
+	 * @param <T>			Type of the message
+	 * @return if parsing successful
+	 */
+	private static <T extends JsonMessage> boolean doReflection(Class<T> clazz, JsonObject jsonObject,
+																Handler<T> handler) {
+		try {
+			var msgClazz = Class.forName(jsonObject.getString("className"));
+			if (!clazz.isAssignableFrom(msgClazz)) {
+				return false;
+			}
+			handler.handle((clazz.cast(jsonObject.mapTo(msgClazz))));
+			return true;
+		} catch (ClassNotFoundException e) {
+			logger.error("Error while converting Json into JsonMessage!", e);
```

Comment:
Please try to avoid the exclamation mark at the end of a log message and be more precise with the message. You need to cover the reason to provide more information. Something like "the requested message class is not available on the classpath".

The error level is way too high here. I think info or warning would be more appropriate. Think about it this way: We expected a certain type of class that is clearly on the classpath but we got a message which is not on the path. So we don't really care about this message. But it could be nice to know, that the system is receiving unknown messages. Maybe trash?

Reply:
Wasn't my code. Thought you did this on purpose :)

Reply:
Then it is better to rework this.